pkg/archive/archive_test.go

```diff
@@ -435,6 +435,34 @@ func TestUntarInvalidFilenames(t *testing.T) {
 	}
 }
 
+func TestUntarHardlinkToSymlink(t *testing.T) {
+	for i, headers := range [][]*tar.Header{
+		{
+			{
+				Name:     "symlink1",
+				Typeflag: tar.TypeSymlink,
+				Linkname: "regfile",
+				Mode:     0644,
+			},
+			{
+				Name:     "symlink2",
+				Typeflag: tar.TypeLink,
+				Linkname: "symlink1",
+				Mode:     0644,
+			},
+			{
+				Name:     "regfile",
+				Typeflag: tar.TypeReg,
+				Mode:     0644,
+			},
+		},
+	} {
+		if err := testBreakout("untar", "docker-TestUntarHardlinkToSymlink", headers); err != nil {
```

Comment:
i'm not sure that you are needing the `testBreakout` helper here, but rather creating a tar with this order of entries, and then `Unpack` the tar to a tmp directory. or by calling `createTarFile()` directly.